_posts/2020-01-01-2019-summary.md

```diff
@@ -0,0 +1,230 @@
+* Reduce query latency and Hive Metastore load.
+* Improve metadata handling during planning. This can result in dramatic improvements in latency, 
+  especially for connectors such as MySQL, PostgreSQL, Redshift, SQL Server, etc. Some queries like 
+  `SHOW SCHEMAS`/`SHOW TABLES` that could take several minutes to complete now finish in a few seconds.
+* Improved stability and performance when spilling is enabled.  
+
+## Functions
+
+* [`combinations`](/docs/current/functions/array.html#combinations)
+* [`format`](/docs/current/functions/conversion.html#format)
+* [`UUID` type](/docs/current/functions/uuid.html) and related functions.
+* [`all_match`](/docs/current/functions/array.html#all_match),
+  [`any_match`](/docs/current/functions/array.html#any_match) and 
+  [`none_match`](/docs/current/functions/array.html#none_match).
+* Support for higher-order [`reduce_agg`](/docs/current/functions/aggregate.html#reduce_agg) 
+  function in aggregation queries.
+* New datetime functions: [`last_day_of_month`](/docs/current/functions/datetime.html#last_day_of_month),
+  [`at_timezone`](/docs/current/functions/datetime.html#at_timezone) and 
+  [`with_timezone`](/docs/current/functions/datetime.html#with_timezone).
+
+## Security
+
+* [Role-based access control](/docs/current/sql/create-role.html) and related commands.
+* [INVOKER security mode](/docs/current/sql/create-view.html#security) for views, which allows views to be run using the permissions of the 
+  current user.
+* Prevent replay attacks and result hijacking in client APIs.
+* JWT-based [internal communication](/docs/current/security/internal-communication.html#internal-authentication) authentication,
+  which obsoletes the need to use Kerberos or certificates and greatly simplifies secure setups.
+* Credential passthrough, which allows Presto to authenticate with the underlying data source with 
+  credentials provided by the user running a query. We found this especially useful when dealing with
+  Google Storage in GCP or SQL databases that manage user authentication and authorization on 
+  their own.
+* Impersonation for [Hive metastore](/docs/current/connector/hive.html#hive-thrift-metastore-configuration-properties).
+* Support for reading and writing encrypted files in HDFS using Hadoop KMS.
+
+## Geospatial
+
+* New geospatial functions: 
+  [`ST_Points`](/docs/current/functions/geospatial.html#ST_Points), 
+  [`ST_Length`](/docs/current/functions/geospatial.html#ST_Length), 
+  [`ST_Area`](/docs/current/functions/geospatial.html#ST_Area), 
+  [`line_interpolate_point`](/docs/current/functions/geospatial.html#line_interpolate_point) and 
+  [`line_interpolate_points`](/docs/current/functions/geospatial.html#line_interpolate_points).
+* `SphericalGeography` type and [related functions](/docs/current/functions/geospatial.html#to_spherical_geography) 
+  to support spatial features in geographic coordinates (latitude / longitude) on spherical model of the earth.
```

Comment:
"on spherical model" sounds wrong grammatically. How about "using a spherical model"

Reply:
That was meant to say "on a spherical model"